extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactivePersistenceProviderSetup.java

```diff
@@ -20,11 +24,22 @@ public static void registerStaticInitPersistenceProvider() {
 
     public static void registerRuntimePersistenceProvider(HibernateOrmRuntimeConfig hibernateOrmRuntimeConfig,
             Map<String, List<HibernateOrmIntegrationRuntimeDescriptor>> integrationRuntimeDescriptors) {
-        jakarta.persistence.spi.PersistenceProviderResolverHolder
-                .setPersistenceProviderResolver(
-                        new SingletonPersistenceProviderResolver(
-                                new FastBootHibernateReactivePersistenceProvider(hibernateOrmRuntimeConfig,
-                                        integrationRuntimeDescriptors)));
+        PersistenceProviderResolver persistenceProviderResolver = PersistenceProviderResolverHolder
+                .getPersistenceProviderResolver();
+        if (persistenceProviderResolver == null ||
+                (persistenceProviderResolver != null
+                        && !(persistenceProviderResolver instanceof MultiplePersistenceProviderResolver))) {
+            persistenceProviderResolver = new MultiplePersistenceProviderResolver();
+            PersistenceProviderResolverHolder.setPersistenceProviderResolver(persistenceProviderResolver);
+        }
+
+        ((MultiplePersistenceProviderResolver) persistenceProviderResolver)
+                .addPersistenceProvider(new FastBootHibernateReactivePersistenceProvider(hibernateOrmRuntimeConfig,
+                        integrationRuntimeDescriptors));
+        ((MultiplePersistenceProviderResolver) persistenceProviderResolver)
+                .addPersistenceProvider(new FastBootHibernatePersistenceProvider(hibernateOrmRuntimeConfig,
+                        integrationRuntimeDescriptors));
```

Comment:
Is that necessary? Wouldn't ORM's own "ProviderSetup" be expected to do it already?